tests/unit/test_model.py

```diff
@@ -335,3 +335,21 @@ def test_model_package_create_transformer_with_product_id(sagemaker_session):
     assert transformer.model_name == 'auto-generated-model'
     assert transformer.instance_type == 'ml.m4.xlarge'
     assert transformer.env is None
+
+
+@patch('sagemaker.fw_utils.tar_and_upload_dir', MagicMock())
+@patch('time.strftime', MagicMock(return_value=TIMESTAMP))
+def test_model_delete_model(sagemaker_session, tmpdir):
+    model = DummyFrameworkModel(sagemaker_session, source_dir=str(tmpdir))
+    model.deploy(instance_type=INSTANCE_TYPE, initial_instance_count=1)
+    model.delete_model()
+
+    sagemaker_session.delete_model.assert_called_with(model.name)
+
+
+@patch('sagemaker.fw_utils.tar_and_upload_dir', MagicMock())
+@patch('time.strftime', MagicMock(return_value=TIMESTAMP))
+def test_delete_non_deployed_model(sagemaker_session, tmpdir):
```

Comment:
Do you need tmpdir or to mock tar_and_upload_dir?

The code path doesn't go through deploy() in this test.